code/src/components/MapMapboxGl.jsx

```diff
@@ -217,6 +217,26 @@ export default class MapMapboxGl extends React.Component {
       });
     })
 
+    map.on("sourcedata", e => {
+      if (!e.isSourceLoaded) {
+        return;
+      }
+
+      let minOrdinal = null;
+      map.queryRenderedFeatures().forEach((f) => {
```

Comment:
not sure if this is the best way to query features. Also tried querySourceFeatures or adding parameters to narrow the query down, but didn't find anything better so far.

Reply:
I am not sure regarding the data flow in the tool, but be aware that queryRenderedFeatures will return no features if the features are filtered out already.

What the indoor module is doing is following: it calculates all sourceLayers from style and aggregates over the source features for all indoor source/source-layers:
https://dev.azure.com/msazure/One/_git/Azure-IoT-Maps-WebSDK?path=/src/IndoorMaps/src/indoor/IndoorManager/IndoorManager.ts&version=GBmaster&line=710&lineEnd=713&lineStartColumn=13&lineEndColumn=75&lineStyle=plain&_a=contents

Reply:
hmm then what if I change it to run this code only if minOrdinal is still null and was not calculated yet? Basically on the initial run only. Would that work or may minOrdinal change later?

Reply:
Sounds great (the initial run) minOrdinal per facility can not change as tilesets are immutable, but it may theoretically change in rare cases if the facility has a weird non-box-like shape when building has a steep incline and we’ve only loaded part of facility on the initial load that doesn’t include the minOrdinal, but that should be rare and you can ignore it for now I think: ref https://www.pinterest.com/pin/236579786655832044/